src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -634,18 +645,22 @@ public ParameterDeclarationSyntax ParseParam(JProperty value)
 
             var typeSyntax = TryParseType(value.Value?["type"]) ?? throw new ConversionFailedException($"Unable to locate 'type' for parameter '{value.Name}'", value);
 
-            if (typeSyntax.TypeName == "securestring")
-            {
-                typeSyntax = new TypeSyntax(SyntaxFactory.CreateToken(TokenType.Identifier, "string"));
-                decoratorsAndNewLines.Add(SyntaxFactory.CreateDecorator("secure"));
-                decoratorsAndNewLines.Add(SyntaxFactory.NewlineToken);
-            }
-
-            if (typeSyntax.TypeName == "secureobject")
+            switch (typeSyntax.TypeName)
             {
-                typeSyntax = new TypeSyntax(SyntaxFactory.CreateToken(TokenType.Identifier, "object"));
-                decoratorsAndNewLines.Add(SyntaxFactory.CreateDecorator("secure"));
-                decoratorsAndNewLines.Add(SyntaxFactory.NewlineToken);
+                case "securestring":
+                    typeSyntax = new TypeSyntax(SyntaxFactory.CreateToken(TokenType.Identifier, "string"));
+                    decoratorsAndNewLines.Add(SyntaxFactory.CreateDecorator("secure"));
+                    decoratorsAndNewLines.Add(SyntaxFactory.NewlineToken);
+                    break;
+                case "secureobject":
+                    typeSyntax = new TypeSyntax(SyntaxFactory.CreateToken(TokenType.Identifier, "object"));
+                    decoratorsAndNewLines.Add(SyntaxFactory.CreateDecorator("secure"));
+                    decoratorsAndNewLines.Add(SyntaxFactory.NewlineToken);
+                    break;
+                case "__bicep_replace":
```

Comment:
>__bicep_replace [](http://example.com/codeflow?start=22&length=15)

When would this be hit? If we can't find an equivalent Bicep type?

Reply:
Where the type isn't known - a variable from the parent template (doesn't have an explicit type) that we're trying to convert to a param (does have an explicit type)